backend/blueprints/tabs.py

```
@@ -27,7 +27,24 @@ def get_tabs():
         A JSON response containing a list of tab objects.
     """
     tabs = Tab.query.order_by(Tab.order).all()
-    return jsonify([tab.to_dict() for tab in tabs])
+
+    # Optimized: Calculate unread counts for all tabs in a single query
+    # to avoid N+1 query problem.
+    unread_counts = (
+        db.session.query(Feed.tab_id, func.count(FeedItem.id))
+        .join(FeedItem, Feed.id == FeedItem.feed_id)
+        .filter(FeedItem.is_read == False)
```

Ruff (0.14.14)

[error] Avoid equality comparisons to `False`; use `not FeedItem.is_read:` for false checks

Replace with `not FeedItem.is_read`

(E712)